### PR TITLE
Explore: Add more meta information when line limit is hit

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**

This PR is part of [logs navigation improvement project](https://www.figma.com/file/7ZHtMoaK4RxBz3b6m6U0cW/Show-more-logs-%2B-Sorting?node-id=611%3A11556). It: 

1. Adds more detailed meta info when we reach line limit. If we won't reach the limit, it shows only information about limit and number received log lines. 
1. Renames **Limit to Line limit** to make it more clear what limit it is. Also I've created Line limit const so if we decide to update naming, it is updated everywhere where used.
1. Removes `Hack to print Loki stats in Explore. Should be using proper stats display via drawer in Explore (rework in 7.1)`. We have full Inspector in Explore and therefore this hack can be removed. 
1. Non related to this PR, but I noticed that border around logs visual preferences was removed. So I ajdusted it based on color/usage at other similar places.
1. Adjusts the font size of labels (they were bigger than everything else) and adds flex-wrap to nicely wrap meta info (instead of smushing everything)

image
image

Related issue:
#19354

### CODE REVIEW

File: public/app/core/logs_model.ts [hunk 1]
```diff
@@ -245,23 +249,33 @@ export function getSeriesProperties(
...
+    const latestTsLogs = sortedRows[sortedRows.length - 1].timeEpochMs;
+    const earliestTsAbsolute = absoluteRange.from;
+    const latestTsAbsolute = absoluteRange.to;
+
+    // Calculate timespan for received logs and range. This is used to calculate timespan coverage meta info of received logs.
+    // We are rounding this as some ds are rounding range in their requests which resulted in incorrect coverage (in edge cases).
+    timeSpanOfLogs = Math.ceil((latestTsLogs - earliestTsLogs) / 1000);
+    timeSpanOfRange = Math.ceil((latestTsAbsolute - earliestTsAbsolute) / 1000);
+
+    const visibleRangeMs = latestTsAbsolute - earliestTsLogs;
```
Comment (ifrost) [inline public/app/core/logs_model.ts:266]: Could it be renamed to `selectedRangeMs`? If I understand correctly it's not the range that is visible on the graph but the range that was selected for the query.

File: public/app/core/logs_model.ts [hunk 2]
```diff
@@ -245,23 +249,33 @@ export function getSeriesProperties(
...
     if (visibleRangeMs > 0) {
```
Comment (ifrost) [inline public/app/core/logs_model.ts:267]: Could `(visibleRangeMs > 0)` be false (it's the selected range now)?

File: public/app/core/logs_model.ts [hunk 3]
```diff
@@ -245,23 +249,33 @@ export function getSeriesProperties(
...
       // Adjust interval bucket size for potentially shorter visible range
       const clampingFactor = visibleRangeMs / (absoluteRange.to - absoluteRange.from);
```
Comment (ifrost) [inline public/app/core/logs_model.ts:269]: Wouldn't it always be `clampingFactor=1` now?

File: public/app/core/logs_model.ts [hunk 4]
```diff
@@ -480,3 +466,48 @@ function getIdField(fieldCache: FieldCache): FieldWithIndex | undefined {
   }
   return undefined;
 }
+
+// Format timeSpan (in sec) to string used in log's meta info
+function formatTimeSpanToTimeString(timeSpanSec: number): string {
```
Comment (ifrost) [inline public/app/core/logs_model.ts:471]: It looks like more accurate version of [secondsToHms()](https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/datetime/rangeutil.ts#L237). Would it make sense to move it to [rangeutil.ts](https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/datetime/rangeutil.ts)?

File: public/app/core/logs_model.ts [hunk 5]
```diff
@@ -480,3 +466,48 @@ function getIdField(fieldCache: FieldCache): FieldWithIndex | undefined {
...
+  (formattedM || formattedH) && formattedS ? (formattedM = formattedM + ' ') : (formattedM = formattedM);
+
+  return formattedH + formattedM + formattedS || 'less than 1sec';
+}
+
+// Used to add additional information to Line limit meta info
+function adjustMetaInfo(logsModel: LogsModel, timeSpanOfLogs?: number, timeSpanOfRange?: number): LogsMetaItem[] {
+  let logsModelMeta = [...logsModel.meta!];
+
+  const limitIndex = logsModelMeta.findIndex((meta) => meta.label === 'Line limit');
```
Comment (ifrost) [inline public/app/core/logs_model.ts:489]: Is there something else we could use to filter? Like an id? Or maybe "Line limit" could be const so nobody changes the label without breaking the filter?

File: public/app/core/logs_model.ts [hunk 6]
```diff
@@ -480,3 +466,48 @@ function getIdField(fieldCache: FieldCache): FieldWithIndex | undefined {
...
+  const limit = limitIndex && logsModelMeta[limitIndex]?.value;
+
+  if (limit && limit > 0) {
+    let metaLimitValue;
+
+    if (limit === logsModel.rows.length && timeSpanOfLogs && timeSpanOfRange) {
+      const coverage = ((timeSpanOfLogs / timeSpanOfRange) * 100).toFixed(2);
+
+      metaLimitValue = `Line limit ${limit} reached, received logs cover ${coverage}% (${formatTimeSpanToTimeString(
```
Comment (ifrost) [inline public/app/core/logs_model.ts:498]: Nit: at the moment it shows: _Line limit: Line limit XXX reached..._ which looks a bit too wordy with "Line limit" repeated. Maybe it could be something like: _Line limit: XXX (max. reached)..._?

File: public/app/core/logs_model.ts (hunk 1 above)
Comment (ivanahuckova) [reply public/app/core/logs_model.ts:266]: This is actually visible range on the Graph, cause you are showing `latestTsAbsolute` and `earliestTsLogs`. It is called visible range across couple of places so I wouldn't change it. However, this is a good point - that instead of using actual log range, we  should use visible range. I'll change it in solution - which will simplify code a bit.

File: public/app/core/logs_model.ts (hunk 2 above)
Comment (ivanahuckova) [reply public/app/core/logs_model.ts:267]: Theoretically it can be. Cause we are (in some DS) rounding up the "to" range when sending request.  And if we receive just 1 log, it can have later time stamp than "to" range.

File: public/app/core/logs_model.ts (hunk 4 above)
Comment (ivanahuckova) [reply public/app/core/logs_model.ts:471]: Yes!

File: public/app/core/logs_model.ts (hunk 5 above)
Comment (ivanahuckova) [reply public/app/core/logs_model.ts:489]: Yeah, we can create const form it.

File: public/app/core/logs_model.ts (hunk 6 above)
Comment (ivanahuckova) [reply public/app/core/logs_model.ts:498]: Make sense. I'll update the copy.

File: public/app/core/logs_model.ts (hunk 4 above)
Comment (ivanahuckova) [reply public/app/core/logs_model.ts]: Moved.

File: public/app/core/logs_model.ts (hunk 3 above)
Comment (ifrost) [reply public/app/core/logs_model.ts:266]: Oh it's been updated. Look good :) I think it could be even be written as:

```suggestion
      const clampingFactor = visibleRangeMs / requestedRangeMs;
```

**Review by ifrost — APPROVED** [review]

**Comment by ifrost** [comment]
yoda
